### PR TITLE
Further align docs for Interchain Queries module

### PR DESCRIPTION
**Pull request by boojamya**
This PR further aligns docs for the new Interchain Queries  app with the upstream IBC-Go repo. 

Changes:
* Adds module to readme
 * Renames the folder `ibc-go/v6/modules/apps/icq` to `ibc-go/v6/modules/apps/32-icq`
 * Removes the App Readme doc from this repo and moves it to the `IBC` repo (currently in a Strangelove fork): https://github.com/strangelove-ventures/ibc/pull/2/files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ In the table below
 
 |Description|Repository|Type|
 |----------|----------|----|
-|An application that enables on chain querying of another IBC enabled chain utilizing `baseapp.Query`. Both chains must have implemented the query application and ICA (for queries requiring consensus).|[ICQ](https://github.com/strangelove-ventures/ibc-go/tree/feature/icq_implementation/modules/apps/icq)|`app`|
+|An application that enables on chain querying of another IBC enabled chain utilizing `baseapp.Query`. Both chains must have implemented the query application and ICA (for queries requiring consensus).|[ICQ](https://github.com/strangelove-ventures/ibc-go/tree/feature/icq_implementation/modules/apps/32-icq)|`app`|
```
Comment (akure) [inline README.md:69]: ICA (for queries requiring consensus -> ICQ (for queries requiring consensus

File: modules/apps/icq/README.md [hunk 2]
```diff
@@ -1,279 +0,0 @@
----
```
Comment (akure) [inline modules/apps/icq/README.md:1]: @boojamya Is this moved to somewhere else? Where are we maintaining the spec doc.

File: README.md (hunk 1 above)
Comment (boojamya) [reply README.md]: Good eye--

File: modules/apps/icq/README.md (hunk 2 above)
Comment (boojamya) [reply modules/apps/icq/README.md:1]: @akure as a follow up, I do think we need to move this over to the `ibc` repo.

**Pull request merged**